docs/DesignDocs/AttestationEvidenceFormatSupport.md

```diff
@@ -0,0 +1,91 @@
+
+Motivation
+----------
+
+Current evidence verify API `oe_verify_evidence()` verifies the evidence gerenated by `oe_get_evidence()` and returns the claims contained in the evidence. Such evidence has a specific format including an `oe_attestation_header`.
+As reported in issue #2999, attestation service has to verify evidences from different sources other than those generated by `oe_get_evidence()`. Such as evidence generated by legacy OE SDK API `oe_get_report()` and raw TEE evidence generation such as SGX quote generation mechanism. And like `oe_verify_evidence()` returns claims, attestation service is also required to return the supplemental information. To achieve that, a new verify API is needed to verify different evidence structures and return the required supplemental information.
+
+Goals:
+- Verifier can call OE SDK API to verify known evidence formats generated by either OE SDK API or other attesters.
+
+User Experience
+---------------
+
+New API will be provided to verify various evidences formats such as SGX quote.
+
+Scenario - Verifier verifies an evidence of known format id:
+
+1. Call OE SDK public API to initialize verifier plugins.
+2. Get an evidence and its format id from attester.
+3. Call the new API with the evidence and the format id to verify the evidence.
+4. The new API returns attestation results and claim set.
+
+
+Specification
+-------------
+
+### New API
+The new API is similar to current function oe_verify_evidence() except the extra `format_id` parameter.
+
+```
+oe_result_t oe_verify_evidence_by_format_id(
+    const oe_uuid_t* format_id,
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    const uint8_t* endorsements_buffer,
+    size_t endorsements_buffer_size,
+    const oe_policy_t* policies,
+    size_t policies_size,
+    oe_claim_t** claims,
+    size_t* claims_length);
+```
+This API is used to verify an evidence whose format id is provided by verifier. It decouples evidence format ID and raw evidence buffer. This API will provide more flexibility.
+
+### New format ID
+ According to issue #2999 reported by attestation service, it has to support evidence generated by SGX quote generation and `oe_get_report()`. SGX quote format id is already defined as `OE_FORMAT_UUID_SGX_ECDSA_P256`. A new id needs to be defined for evidence generated by `oe_get_report()`.
+
+```
+#define OE_FORMAT_UUID_OE_REPORT { _uuid_ }
+```
+
+### Additional Claims
+Without custom claims, a new `oe_claim` field is introduced to retrieve `report_data` field inside sgx_quote. Attestation service needs this field as a part of their validation process.
+
+```
+#define OE_CLAIM_REPORT_DATA "report_data"
+```
+
+Alternates
+----------
+We can also develop APIs that convert a raw evidence to and from an oe_evidence.
+
+### convert_to_evidence().
```

Comment:
My opinion: I don't think we need this API, we can abandon the oe_evidence going forward.